### PR TITLE
firefox/wrapper: clean up plugin related code

### PR DESCRIPTION
**Pull request by xaverdh**
###### Motivation for this change
Some more cleanup after https://github.com/NixOS/nixpkgs/pull/112335.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xaverdh** [comment]
Does the nix language have something like `assert` with customizable error message?
I searched for something like that, but couldn't find anything..

File: pkgs/applications/networking/browsers/firefox/wrapper.nix [hunk 1]
```diff
@@ -164,7 +143,25 @@ let
       #                           #
       #############################
 
-    in stdenv.mkDerivation {
+      # TODO: remove this in the future
```
Comment (xaverdh) [inline pkgs/applications/networking/browsers/firefox/wrapper.nix:146]: when do we want to remove this actually? after branch off (of 21.03) or the release after that maybe?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112491` at 5b86a260 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>dropbox</li>
    <li><a href="https://gist.github.com/e4c7b92aa759b43bef621790409be983">dropbox-cli</a></li></li>
    <li><a href="https://gist.github.com/7835ee68850ea36694c7a9a05004244f">mate.caja-dropbox</a></li></li>
    <li><a href="https://gist.github.com/ed28890ad922bb7c3cae6b67b20df98b">slimerjs</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112491` at 5b86a260 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>slimerjs</li>
  </ul>
</details>

File: pkgs/applications/networking/browsers/firefox/wrapper.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:146]: After the next release sounds reasonable.

**Comment by bjornfor** [comment]
> Does the nix language have something like `assert` with customizable error message?
> I searched for something like that, but couldn't find anything..

Did you try 'if cond then throw "msg" else thing'?

**Comment by rmcgibbo** [comment]
Maybe `lib.assertMsg` is what you want? https://github.com/NixOS/nixpkgs/blob/master/lib/asserts.nix#L21

**Comment by xaverdh** [comment]
Well assert has some special syntax which avoids the braces in the following expression.
But `if then else` is a good suggestion. Its slightly more readable than using `seq` I think.

**Review by taku0 — APPROVED** [review]

**Pull request merged**